libraries/shared/storage/backfill.go

```diff
@@ -0,0 +1,125 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package storage
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/statediff"
+	"github.com/sirupsen/logrus"
+
+	"github.com/vulcanize/vulcanizedb/libraries/shared/fetcher"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/storage/utils"
+)
+
+// IBackFiller is the backfilling interface
+type IBackFiller interface {
```

Comment:
nbd but I've been trying to think of ways we can avoid names like this (`Ix` or `xInterface`) and one option that's seemed attractive to me is to name the interface something like `BackFiller` and then not exporting the implementation (e.g. struct becomes `backFiller`). The linter will complain if the constructor returns a non-exported struct but usually you can have a constructor return the interface

Reply:
I also prefer the unexported struct approach, the one potential hangup I've ran into with this (other than the linter) is if in tests we want to work with the underlying struct instead of the interface so that we can insert mocks or otherwise modify the underlying struct directly for testing purposes.